monai/data/image_reader.py

```diff
@@ -226,64 +238,55 @@ def _get_meta_dict(self, img) -> Dict:
         Get all the meta data of the image and convert to dict type.
 
         Args:
-            img: a ITK image object loaded from a image file.
+            img: an ITK image object loaded from an image file.
 
         """
         img_meta_dict = img.GetMetaDataDictionary()
-        meta_dict = {}
-        for key in img_meta_dict.GetKeys():
-            # ignore deprecated, legacy members that cause issues
-            if key.startswith("ITK_original_"):
-                continue
-            if (
-                key == "NRRD_measurement frame"
-                and int(itk.Version.GetITKMajorVersion()) == 5
-                and int(itk.Version.GetITKMinorVersion()) < 2
-            ):
-                warnings.warn(
-                    "Ignoring 'measurement frame' field. "
-                    "Correct reading of NRRD05 files requires ITK >= 5.2: `pip install --upgrade --pre itk`"
-                )
-                continue
-            meta_dict[key] = img_meta_dict[key]
-        meta_dict["origin"] = np.asarray(img.GetOrigin())
+        meta_dict = {key: img_meta_dict[key] for key in img_meta_dict.GetKeys() if not key.startswith("ITK_")}
+
         meta_dict["spacing"] = np.asarray(img.GetSpacing())
-        meta_dict["direction"] = itk.array_from_matrix(img.GetDirection())
         return meta_dict
 
     def _get_affine(self, img):
         """
         Get or construct the affine matrix of the image, it can be used to correct
         spacing, orientation or execute spatial transforms.
-        Construct Affine matrix based on direction, spacing, origin information.
-        Refer to: https://github.com/RSIP-Vision/medio
 
         Args:
-            img: a ITK image object loaded from a image file.
+            img: an ITK image object loaded from an image file.
 
         """
         direction = itk.array_from_matrix(img.GetDirection())
         spacing = np.asarray(img.GetSpacing())
         origin = np.asarray(img.GetOrigin())
 
         direction = np.asarray(direction)
-        affine: np.ndarray = np.eye(direction.shape[0] + 1)
-        affine[(slice(-1), slice(-1))] = direction @ np.diag(spacing)
-        affine[(slice(-1), -1)] = origin
+        sr = min(max(direction.shape[0], 1), 3)
+        affine: np.ndarray = np.eye(sr + 1)
+        affine[:sr, :sr] = direction[:sr, :sr] @ np.diag(spacing[:sr])
+        affine[:sr, -1] = origin[:sr]
+        flip_diag = [[-1, 1], [-1, -1, 1], [-1, -1, 1, 1]][sr - 1]
```

Comment:
Could you please help add a comment or doc-string here? in case we forget the logic in future..